### ISSUE TITLE
New reciter - Ali Jaber

### ISSUE THREAD

**Issue by MJayyusi** [quran/quran_android#1329]
Ali Jaber Quran Audio can be found under https://download.quranicaudio.com/quran/abdullaah_alee_jaabir_studio/

I will depend on verse by verse timings (with some time adjustments) to figure out the proper timing db.

**Comment by MoeHus** [comment]
Can you please assign this issue to me if this sounds proper. 
Thank you.

**Comment by ahmedre** [comment]
wa3laikum alsalam,
please, would love to have this! how are you planning on doing it (perhaps I can point you to some things to save some time for you).
jazakAllah khairan.
walsalam 3alaikum.

**Comment by MJayyusi** [comment]
I opened up the verse by verse version and looked at the audio waves, then noticed its a compressed version of the gapless version. The compression affected the quality where it can't be properly matched through tools like Dejavu.

By looking at them, I noticed only the Basmalah is different and its position needs to be manually indicated, and that there is additional 50 milliseconds shift after each Ayah. 
I developed a short script to generate the timings based on this info. 
So far I tried it for Al baqarah, Al-Imran and Al Tawbah, that worked perfectly. 

I am aware of the Quran Timing repo by Anassaiyed, that didn't work for this case.

**Comment by ahmedre** [comment]
masha'Allah, amazing! would love it if you can give a database, lots of people ask for sheikh Ali Jaber! jazakAllah khairan.

if you end up having to do any mp3s completely manually, let me know since I built a python script that makes it easier to listen to and find the ayah gaps without going through the entire file.

**Comment by MJayyusi** [comment]
Salam @ahmedre 
I am hitting a wall here. I noticed there are some defects here and there in the audio recording, all were minor until I reached Surat Al Noor! the recording I found (in the provided link above) is only about 8 min long! until Ayah 27
I looked at the verse by verse recoding and its all good. so there must be a better gapless version of this. Can you please help.

**Comment by MJayyusi** [comment]
@ahmedre 
Alright I found two copies but not sure which one to go with. Would please take a look 
http://www.alijaber.net/murattal/?dir=1-FLAT
http://www.alijaber.net/murattal/?dir=3-alijaberNet

**Comment by ahmedre** [comment]
I can't actually access it right now (it's not loading for me) - but in sha' Allah can try these. if they are high quality and have more correct recordings, let's use them (we can replace the quranicaudio one with this one if so). if the issues in quranicaudio are minor (or if the quality of theirs isn't very high), we can also fix the ones one quranicaudio (and replace the broken sura only for example).

**Comment by MJayyusi** [comment]
Salam @ahmedre  I emailed you at quran.android. 
I need your help. Thank you.

**Comment by MJayyusi** [comment]
Salam @ahmedre 
Can you please place the uploaded files at their appropriate locations. 
Thank you
Moe

**Comment by ahmedre** [comment]
wa3laikum alsalam,
uploaded audio files here:
https://download.quranicaudio.com/quran/abdullaah_alee_jaabir_studio/re-encoded - guessing these are exactly the same files as https://download.quranicaudio.com/quran/abdullaah_alee_jaabir_studio/ but just re-encoded so that the timings end up correct on the apps, right?

also, would you be ok if i changed the id3 tags (just the image to use quranicaudio logo instead of the sheikh's image)?

uploaded the database as ali_jaber.zip as well.
jazakAllah khairan!
walsalam 3alaikum.

**Comment by ahmedre** [comment]
just wanted to say masha'Allah amazing job! here's an apk to test with:
http://cafesalam.net/uploads/sheikh_ali_jaber.apk

**Comment by nacer80** [comment]
Salam 3alikom  
Great job my brothers
Sadaka jarya Inshallah 
Jazakoum Allaho kheiren

**Comment by MJayyusi** [comment]
@ahmedre Sorry, I couldn't respond any earlier. This isn't a reencoded audio of the link you provided. 
I copied the original files from http://www.alijaber.net/murattal/?dir=3-alijaberNet
I did very few tweaks beside the reencoding, which included removing some repeated ayas (errors in the original recordings)

Thank you , though you took the pleasure of providing the rest of the code changes :P

I still don't understand where do you keep the links for the audio files and timing dbs. do you use an online lookup table of some sort?

I'd like to participate future in supporting Android Auto and Google Chromecast, but yet I have to understand how everything works. 

I had to build only madani edition of the app to test the timing db for this qari.

**Comment by ahmedre** [comment]
do you think the directory doesn't make sense - should i copy it somewhere else on the server instead of /reencoded/? i figured they were the same source at least because the duration of sura Baqarah matched.

for reading the data, i have a full version of:
https://github.com/quran/quran_android/blob/master/pages/madani/src/main/res/values/readers.xml

that i replace when building the release. i used to keep the whole file open, but people would just clone the app as it is, add ads and push it.

yes if it works on madani it should work on Naskh/Shemerly - the others (Warsh/Qaloon) use different sets of reciters because of recitation differences.

jazakAllah khairan and looking forward to more contributions in the future!

**Comment by MJayyusi** [comment]
It is different. I think that one is a very different quality too, its missing most of Surat Al Noor. 

I guess its better to differentiate in case someone is trying to download a copy or trying to cleanup in the future, just so they don't get confused. But you know better than me, I am new to this. 

Thank you for the explanation about readers.xml now it makes senses.

**Comment by MJayyusi** [comment]
@ahmedre 
If you are changing the path please rename it to ali_jaber or any proper name spelling it was very tough to find with the existing spelling.

**Comment by ahmedre** [comment]
done - it's now https://download.quranicaudio.com/quran/ali_jaber/
fixed also the apk - here's the updated link: http://cafesalam.net/uploads/sheikh_ali_jaber.apk

btw can you help me with one small thing - now we have 3 copies on the site:
https://download.quranicaudio.com/quran/abdullaah_alee_jaabir_studio and
https://download.quranicaudio.com/quran/abdullaah_alee_jaabir (this sounds like non-studio)

any recommendations on removing one of them or does it make sense to have all 3? at least if ali_jaber and abdullaah_alee_jaabir_studio are similar, should i try to remove abdullaah_alee_jaabir_studio?

also, do you mind if i change the id3tag image to the quranicaudio.com logo?

jazakAllah khairan!

**Comment by MJayyusi** [comment]
Yes you can change album art or remove it, just keep the other data (Surah name), feel free to make changes as needed. 

I'll analyze the other two directories tonight and let you know.

**Comment by MJayyusi** [comment]
Salam brother @ahmedre 
I looked up the audio files for Ali Jaber, the old ones you have are lower quality and one incomplete surah. 
The ones I have are much better. 
Below are the stream attributes for each one
Master 
320 kb/s 48.0 kHz
reencoded to 
128 kb/s 48.0 kHz

abdullaah_alee_jaabir and Studio 
128 kb/s 44.1 kHz

Though there isn't much difference to human ears between 44.1 kHz and 48.0 kHz, the ones I provided are better recordings in general.

I recommend deleting abdullaah_alee_jaabir and abdullaah_alee_jaabir_studio

And keeping the ones uploaded <link below> 
(Master is the original copy, its much larger but good to keep for editing purposes)

https://drive.google.com/file/d/1EzlHUaTx5egwcvevVH5cWNxdDFA79jiS/view?usp=sharing
